src/zarr/abc/metadata.py

```diff
@@ -29,6 +30,8 @@ def to_dict(self) -> dict[str, JSON]:
             value = getattr(self, key)
             if isinstance(value, Metadata):
                 out_dict[field.name] = getattr(self, field.name).to_dict()
+            elif isinstance(value, bytes):
+                out_dict[key] = base64.b64encode(value)
```

Comment:
In general, we can't assume bytes are valid JSON so they must be transformed somehow. base64 encoding seems as good as anything else.

We'll need to update the `from_dict` to do the decoding. We probably need to see that we're decoding to a bytes dtype and use that when parsing the fill value.